src/main/resources/com/google/api/codegen/js/sample.snip

```diff
@@ -0,0 +1,181 @@
+
+    @end
+    @if class.isPageStreaming
+      {@pageStreamingFuncSignature(class)};
+    @else
+      var {@class.requestVarName} = {@requestFuncSignature(class)};
+      @if class.hasResponse
+        {@class.requestVarName}.then(function({@class.responseVarName}) {
+          // TODO: Change code below to process the `{@class.responseVarName}` object:
+          console.log({@class.responseVarName}.result);
+        }, function(reason) {
+          console.log('error: ' + reason.result.error.message);
+        });
+      @else
+        {@class.requestVarName}.then(function() {}, function(reason) {
+          console.log('error: ' + reason.result.error.message');
+        });
+      @end
+    @end
+  }
+
+  @if class.isPageStreaming
+    function {@pageStreamingFuncSignature(class)} {
+      var {@class.requestVarName} = {@requestFuncSignature(class)};
+      request.then(function({@class.responseVarName}) {
+        var {@class.pageStreaming.pageVarName} = {@class.responseVarName}.result.{@class.pageStreaming.resourceFieldName};
+        @if class.pageStreaming.isResourceRepeated
+          @if class.pageStreaming.isResourceMap
+            Object.keys({@class.pageStreaming.pageVarName}).forEach(function({@class.pageStreaming.resourceKeyVarName}) {
+              // TODO: Change code below to process each `{@class.pageStreaming.resourceKeyVarName}` property:
+              console.log({@class.pageStreaming.resourceKeyVarName} + ': ' + JSON.stringify({@class.pageStreaming.pageVarName}[{@class.pageStreaming.resourceKeyVarName}], null, 2));
+            });
+          @else
+            for (var i = 0; i < {@class.pageStreaming.pageVarName}.length; i++) {
+              // TODO: Change code below to process each resource in `{@class.pageStreaming.pageVarName}`:
+              console.log(JSON.stringify({@class.pageStreaming.pageVarName}[i], null, 2));
+            }
+          @end
+        @else
+          // TODO: Change code below to process each `{@class.pageStreaming.pageVarName}` resource:
+          console.log({@class.pageStreaming.pageVarName});
+        @end
+
+        if ({@class.responseVarName}.nextPageToken) {
+          @if class.pageStreaming.isResourceSetterInRequestBody
+            {@class.requestBodyVarName}.pageToken = {@class.responseVarName()}.nextPageToken;
+          @else
+            {@class.paramsVarName}.pageToken = {@class.responseVarName()}.nextPageToken;
+          @end
+          {@pageStreamingFuncSignature(class)};
+        }
+      }, function(reason) {
+        console.log('error: ' + reason.result.error.message);
+      });
+    }
+
+  @end
+  function start() {
+    gapi.client.init({
+      @switch class.auth.type
+      @case "API_KEY"
+        'apiKey': '',  // TODO: Update placeholder with desired API key.'
+      @default
+        'clientId': '',  // TODO: Update placeholder with desired client ID.
+        'scope': '{@scopeString(class.auth.scopes)}',
```

Comment:
It's... not necessarily the case. It's possible that an API could require more than one scope to perform a function (there's a sentence in one of the docs about this, I can find it if you like), but most require only one. The trick becomes deciding which one to include... in the other languages we don't implement the OAuth2 flow and don't make a decision, so here I felt it was better to err towards including everything so the request works.

We don't actually have enough context to make an informed decision, API explorer also checks all scopes that a method declares.